### PR TITLE
add requirements.txt, use in CI

### PR DESCRIPTION
**Pull request by tjhei**
- add basic requirements.txt
- update readme
- use requirements.txt in CI

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -23,7 +23,8 @@ jobs:
          texlive \
          texlive-latex-extra
         python -m pip install --upgrade pip
-        pip install matplotlib numpy scipy sympy pycddlib cvxpy
+        python -m pip install -r requirements.txt
```
Comment (bobmyhill) [inline .github/workflows/main.yml:26]: we already have sphinx/pip_requirements.txt.

I'd suggest that we remove that file, rename the new requirements.txt to pip_requirements.txt, and change the line in test.sh from
`$PYTHON -m pip freeze | grep -f sphinx/pip_requirements.txt`
to
`$PYTHON -m pip freeze | grep -f ./pip_requirements.txt`

**Comment by tjhei** [comment]
You know that ./requirements.txt is a common if not standard way to do things, right? The sphinx requirements are different, do you really want to merge them?

**Comment by bobmyhill** [comment]
Yes, I know that a requirements.txt file is a standard way to check requirements (this is why I added the sphinx/pip_requirements.txt file back in 2017). The existing file is only used by ReadTheDocs. 

I'm not sure it would matter if we had all the requirements (code + docs) in a single file, but if you'd like to keep two separate files, I'm not going to insist on a merger. I just thought it would be nice to minimise file duplication.

**Comment by tjhei** [comment]
I don't feel strongly about it, so feel free to change things!

**Pull request merged**